aptos-move/framework/aptos-framework/sources/aptos_governance.move

```diff
@@ -334,6 +397,22 @@ module aptos_framework::aptos_governance {
             should_pass,
         );
 
+        let record_key = RecordKey {
+            stake_pool,
+            proposal_id,
+        };
+        if (features::partial_governance_voting_enabled()) {
+            let voting_records_v2 = borrow_global_mut<VotingRecordsV2>(@aptos_framework);
+            let used_voting_power = table::borrow_mut_with_default(&mut voting_records_v2.votes, record_key, 0);
+            *used_voting_power = *used_voting_power + voting_power;
+        } else {
```

Comment:
This logic is a bit hard to follow in the case of a proposal created before partial voting is enabled and with voting extending beyond. IMO, it's likely cleaner to fail early if VotingRecords is already updated and:
1. Update VotingRecords if voting power is MAX_U64
2. Update VotingRecordsV2 if partial voting is enabled, regardless of if voting power is MAX_U64 or not.